windows/src/desktop/kmshell/main/OnlineUpdateCheck.pas

```diff
@@ -339,6 +463,128 @@ function TOnlineUpdateCheck.DownloadUpdates: Boolean;
     else DeleteFileOnReboot(ExcludeTrailingPathDelimiter(DownloadTempPath));
 end;
 
+
+procedure TOnlineUpdateCheck.DoDownloadUpdatesBackground(SavePath: string; var Result: Boolean);
+var
+  i, downloadCount: Integer;
+
+    function DownloadFile(const url, savepath: string): Boolean;
+    begin
+      Result := False;
+      with THttpUploader.Create(nil) do
```

Comment:
We need to check also if THttpUploader is allowed to run in a non-gui process. Some Windows internet functions require gui, others are allowed to run in non-gui processes -- and I can't remember which is which at present.

Reply:
Yeah, as I vaguely remembered... `THttpUploader` uses `InternetOpen` and pals. Per [Microsoft documentation](https://learn.microsoft.com/en-us/windows/win32/api/wininet/nf-wininet-internetopena):

> Note  WinINet does not support server implementations. In addition, it should not be used from a service. For server implementations or services use [Microsoft Windows HTTP Services (WinHTTP)](https://learn.microsoft.com/en-us/windows/desktop/WinHttp/winhttp-start-page).

Given our desire to move this to a service model, we probably need to rewrite the downloader.